src/RecordData/psram.h

```diff
@@ -1,156 +1,50 @@
 #ifndef PSRAM_H
 #define PSRAM_H
 
-#include "StorageType.h"
 #include <cstdint>
+#include "../Constants.h"
 
-namespace mmfs {
-
-extern "C" uint8_t external_psram_size;
-class PSRAM: public StorageType
+extern "C" uint8_t external_psram_size; // in MB
+namespace mmfs
 {
+    class PSRAMFile;
+    class PSRAM
+    {
 
-private:
-    bool ready;
-    bool dumped;
-    char *cursorStart; // for flight data
-    char *cursorEnd; // for log data
-    char *memBegin; // start of memory
-    char *memEnd; // end of memory
-
-public:
-    /**
-     * Constructor for PSRAM.
-     */
-    PSRAM();
-
-    /**
-     * Initialize the PSRAM.
-     * @return true if initialization was successful, false otherwise.
-     */
-    bool init() override;
-
-    /**
-     * Check if the PSRAM is ready for read/write operations.
-     * @return true if the PSRAM is ready, false otherwise.
-     */
-    bool isReady() const override;
-
-    /**
-     * Write data to the PSRAM.
-     * @param data The data to be written.
-     * @param size The size of the data to be written.
-     */
-    void write(char *data, int size) override;
-
-    /**
-     * Write data to the PSRAM with an option to write to the top.
-     * @param data The data to be written.
-     * @param size The size of the data to be written.
-     * @param writeToTop If true, write to the top of the memory. If false, write to the bottom in reverse.
-     */
-    void write(char *data, int size, bool writeToTop);
-
-    /**
-     * Print a string to the PSRAM.
-     * @param data The string to be printed. Takes a pointer to a null-terminated character array.
-     */
-    void print(const char *data) { print(data, true); };
-
-    /**
-     * Print a string to the PSRAM with an option to write to the top.
-     * @param data The string to be printed. Takes a pointer to a null-terminated character array.
-     * @param writeToTop If true, write to the top of the memory. If false, write to the bottom in reverse.
-     */
-    void print(const char *data, bool writeToTop);
-
-    /**
-     * Print a string to the PSRAM followed by a newline.
-     * @param data The string to be printed. Takes a pointer to a null-terminated character array.
-     */
-    void println(const char *data) { println(data, true); };
+        friend class PSRAMFile;
 
-    /**
-     * Print a string to the PSRAM followed by a newline with an option to write to the top.
-     * @param data The string to be printed.
-     * @param writeToTop If true, write to the top of the memory. If false, write to the bottom in reverse.
-     */
-    void println(const char *data, bool writeToTop);
+        inline int min(int a, int b) { return a < b ? a : b; }
+        inline int max(int a, int b) { return a > b ? a : b; }
```

Comment:
doesn't Arduino and/or C++11 have this built-in? Is this just to be platform-agnostic

Reply:
yeah exactly

Reply:
It was needed for testing on native but instead of making this file reliant on arduino.h I just put these two small functions here